src/java.base/share/classes/sun/security/ssl/SSLSessionImpl.java

```diff
@@ -501,8 +500,13 @@ final class SSLSessionImpl extends ExtendedSSLSession {
                 buf.get(b);
                 this.preSharedKey = new SecretKeySpec(b, alg);
                 // Get identity len
-                this.pskIdentity = new byte[buf.get()];
-                buf.get(pskIdentity);
+                i = buf.get();
```

Comment:
Given that this is just a small change to a large function it may not be important to change, but a lot of these X-byte lengths followed by arrays might be simplified with Record.getBytes8 or the other similar routines that handle longer length fields.  But you wouldn't end up with a null byte[] when the length is zero, so that's a change, not necessarily a down-side.

Reply:
I think that's something to consider as a new RFE as it would have a scope outside of this change